src/main/java/com/minecolonies/coremod/entity/citizenhandlers/CitizenHappinessHandler.java

```diff
@@ -0,0 +1,586 @@
+package com.minecolonies.coremod.entity.citizenhandlers;
+
+import com.minecolonies.api.util.constant.CitizenConstants;
+import com.minecolonies.api.util.constant.IToolType;
+import com.minecolonies.api.util.constant.ToolType;
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.FieldDataModifier;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import com.minecolonies.coremod.entity.ai.util.ChatSpamFilter;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.common.util.Constants;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Handler for the citizens happiness.
+ * This keeps all the modifiers related to the citizen.
+ * 
+ * @author kevin
+ *
+ */
+public class CitizenHappinessHandler
+{
+
+    @NotNull
+    protected final ChatSpamFilter chatSpamFilter;
+
+    /**
+     * Tag names for NBT fields
+     */
+    private static final String TAG_NAME = "happiness";
+    private static final String TAG_BASE = "base";
+    private static final String TAG_FOOD = "foodModifier";
+    private static final String TAG_DAMAGE = "damageModifier";
+    private static final String TAG_HOUSE = "houseModifier";
```

Comment:
I would put these constants into the constant files.

Reply:
Why would you do this.   From what I've seen all classes that define constant for NBT files are done inside the class it is using it.  If you put it in the constant files, it makes it global?

Reply:
We started moving a bunch of these over because having a lot of constants within one class clutters the class a bit + most constants we will be able to reuse in other occasions.

Reply:
okay